packages/lesmis/src/views/MapView.js

```diff
@@ -17,27 +33,137 @@ const Container = styled.div`
   min-height: 600px;
 `;
 
+const Map = styled(MapContainer)`
+  flex: 1;
+  height: 100%;
+`;
+
 const Main = styled.div`
   position: relative;
   flex: 1;
   height: 100%;
 `;
 
+const EmptyPanel = styled(MapOverlaysPanelContainer)`
+  padding: 2rem;
+`;
+
+const Legend = styled(LegendComponent)`
+  position: absolute;
+  bottom: 5px;
+  z-index: 999;
+  left: 50%;
+  transform: translateX(-50%);
+`;
+
+const TilePicker = styled(TilePickerComponent)`
+  position: absolute;
+  right: 0;
+  bottom: 0;
+  z-index: 9999;
+  pointer-events: none;
+`;
+
+const useProvinceData = () => {
+  const query = useEntitiesData();
+  let provinces = [];
+
+  if (query.data) {
+    provinces = query.data.filter(entity => entity.parentCode === 'LA');
+  }
+
+  return { ...query, data: provinces };
+};
+
 export const MapView = () => {
   const { entityCode } = useUrlParams();
+  const [activeTileSetKey, setActiveTileSetKey] = useState(TILE_SETS[0].key);
+  const [hiddenMeasures, setHiddenMeasures] = useState([]);
   const [selectedOverlay, setSelectedOverlay] = useState(null);
 
+  useEffect(() => {
+    setHiddenMeasures([]);
+  }, [setHiddenMeasures, selectedOverlay]);
+
+  const { data: entityData } = useEntityData({
```

Comment:
It might make the component a little simpler to pull these four hooks out into one `useMapViewData` that has a combined loading state along with each of the pieces of data as keys in the return object

Reply:
In the end I didn't pull all the hooks together but I did reduce it down to just 2 - one for the overlay selector and one for the overlays